src/libs/elektra/key.c

```diff
@@ -626,3 +764,84 @@ ssize_t keyGetRef (const Key * key)
 
 	return key->ksReference;
 }
+
+
+/**
+ * @internal
+ * @see keyLock + keyIsLocked
+ */
+static int elektraKeyLock (Key * key, /*option_t*/ enum elektraLockOptions what, int test)
+{
+	int ret = 0;
+
+	if (!key) return -1;
+
+	if (test_bit (what, KEY_LOCK_NAME))
+	{
+		if (!test_bit (key->flags, KEY_FLAG_RO_NAME))
+		{
+			if (!test) set_bit (key->flags, KEY_FLAG_RO_NAME);
+			set_bit (ret, KEY_LOCK_NAME);
+		}
+	}
+
+	if (test_bit (what, KEY_LOCK_VALUE))
+	{
+		if (!test_bit (key->flags, KEY_FLAG_RO_VALUE))
+		{
+			if (!test) set_bit (key->flags, KEY_FLAG_RO_VALUE);
+			set_bit (ret, KEY_LOCK_VALUE);
+		}
+	}
+
+	if (test_bit (what, KEY_LOCK_META))
+	{
+		if (!test_bit (key->flags, KEY_FLAG_RO_META))
+		{
+			if (!test) set_bit (key->flags, KEY_FLAG_RO_META);
+			set_bit (ret, KEY_LOCK_META);
+		}
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Permanently locks a part of the key
+ *
+ * This can be:
+ * - KEY_LOCK_NAME to lock the name
+ * - KEY_LOCK_VALUE to lock the value
+ * - KEY_LOCK_META to lock the metadata
+ *
+ * To unlock the key, duplicate it.
+ *
+ * It is also possible to lock when the key is created with
+ * keyNew().
+ *
+ * Some data structures need to lock the key (most likely
+ * its name), so that the ordering does not get confused.
+ *
+ * @param key which name should be locked
+ *
+ * @see keyNew(), keyDup(), ksAppendKey()
+ * @retval >0 the bits that were successfully locked
+ * @retval 0 if everything was locked before
+ * @retval -1 if it could not be locked (nullpointer)
+ * @ingroup key
+ */
+int keyLock (Key * key, enum elektraLockOptions what)
+{
+	return elektraKeyLock (key, what, 0);
+}
+
+/**
+ * @brief Tests if a part of the key can or has been locked
```

Comment:
Unclear what "can be locked" is (and how to find out about it).

Reply:
Please provide me with the complete description. I don't need any @brief at all.

Reply:
"Tests if a part of a key is locked" should be fine?